lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -28,6 +28,26 @@ defmodule Kernel.WarningTest do
     purge Sample
   end
 
+  test "unused variable in redefined function in different file" do
+    output = capture_err(fn ->
+      Code.eval_string """
+      defmodule Sample do
```

Comment:
Not sure I understand what you mean by "use the supervisor example". I can't find an existing fixture that has redefinition of function from a different source. I can avoid the eval_string by change the fixture file to ```use supervisor```, but this would still require having a fixture file.